stan/math/prim/mat/fun/to_matrix.hpp

```diff
@@ -24,65 +29,47 @@ namespace stan {
     }
 
     /**
-     * Returns a matrix representation of the vector in column-major order
-     * with the specified number of rows and columns.
+     * Returns a matrix representation of a standard vector of Eigen
+     * row vectors with the same dimensions and indexing order.
      *
      * @tparam T type of the scalar
-     * @param x vector of values
-     * @param m rows
-     * @param n columns
-     * @return the matrix representation of the input
-     * @throw <code>std::invalid_argument</code> if the sizes do not match
-     */
-    template <typename T>
-    inline Eigen::Matrix<T, Eigen::Dynamic, Eigen::Dynamic>
-    to_matrix(const std::vector<T>& x, int m, int n) {
-      static const char* fun = "to_matrix(array)";
-      check_size_match(fun, "rows * columns", m * n, "vector size", x.size());
-      return Eigen::Map<const Eigen::Matrix<T, Eigen::Dynamic,
-                                            Eigen::Dynamic> >(&x[0], m, n);
-    }
-
-    /**
-     * Returns a matrix representation of the standard vector of standard vectors
-     * with the same dimensions and indexing order.
-     *
-     * @tparam T type of the scalar
-     * @param x vector of vectors of scalar values
+     * @param x Eigen vector of vectors of scalar values
      * @return the matrix representation of the input
      */
     template <typename T>
     inline Eigen::Matrix<T, Eigen::Dynamic, Eigen::Dynamic>
-    to_matrix(const std::vector< std::vector<T> >& x) {
-      size_t rows = x.size();
+    to_matrix(const
+              std::vector<Eigen::Matrix<T, 1, Eigen::Dynamic> >& x) {
+      int rows = x.size();
       if (rows != 0) {
-        size_t cols = x[0].size();
-        Eigen::Matrix<T, Eigen::Dynamic, Eigen::Dynamic> result(rows, cols);
-        for (size_t i=0, ij=0; i < cols; i++)
-          for (size_t j=0; j < rows; j++, ij++)
+        int cols = x[0].size();
+        Eigen::Matrix<T, Eigen::Dynamic, Eigen::Dynamic>
+          result(rows, cols);
+        for (int i=0, ij=0; i < cols; i++)
+          for (int j=0; j < rows; j++, ij++)
             result(ij) = x[j][i];
         return result;
       } else {
         return Eigen::Matrix<T, Eigen::Dynamic, Eigen::Dynamic> (0, 0);
```

Comment:
Since rows and cols are used multiple times (on each iteration of the loop), having them defined locally does speed up thing, doesn't it?